contracts/libraries/SafeTransferLib.sol

```
@@ -9,6 +9,24 @@ import {Errors} from "@libraries/Errors.sol";
 /// @author Modified from Solmate (https://github.com/Rari-Capital/solmate/blob/main/src/utils/SafeTransferLib.sol)
 /// @dev Caution! This library won't check that a token has code, responsibility is delegated to the caller.
 library SafeTransferLib {
+    /*//////////////////////////////////////////////////////////////
+                             ETH OPERATIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Safely transfers ETH to a specified address.
+    /// @param to The address to transfer ETH to
+    /// @param amount The amount of ETH to transfer
+    function safeTransferETH(address to, uint256 amount) internal {
```

```suggestion
    function safeTransferETH(address to, uint256 amount) internal {
        if (amount) {
            bool success;

            assembly {
                // Transfer the ETH and store if it succeeded or not.
                success := call(gas(), to, amount, 0, 0, 0, 0)
            }

            if (!success) revert Errors.TransferFailed();
        }
    }
```

You could also consider moving your `surplus > 0` check to this method and that way you wouldn't need the `surplus` var in `CollateralTracker.unlockCallback`.  This version fails silently if `amount` is 0, which plays nicely with `unlockCallback`'s needs, but could be unintuitive to other users of this helper though - they might expect a proper revert

if you adopted this i'd probably also modify the docstring:

```
/// @notice Safely transfers a positive amount of ETH to a specified address.
/// @param to The address to transfer ETH to
/// @param amount The amount of ETH to transfer
```

---

> if (amount) {

There is no way to cast anything to bool type in solidity without using comparison operators unfortunately :(

> way you wouldn't need the surplus var in CollateralTracker.unlockCallback.

When you see variables being assigned to an expression in code, that doesn't necessarily mean that the gas cost increases significantly over the expression itself, it just means that the stack item is duplicated and persisted when it gets use within its scope.

i.e. the current sequence of opcodes would be something like SUB DUP1 PUSH0 LT PUSHN jmpdest JUMPI
and the new one would be SUB PUSHN jmpdest JUMP DUP1 PUSH0 LT PUSHN jmpdest JUMPI (jumpback logic based on that if statement...), which is not a clear improvement and may even cost more gas (I have of course omitted stack management for the other variables).

ceteris paribus I think the current version is a little more explicit about the behavior on different `surplus` values

---

> When you see variables being assigned to an expression in code, that doesn't necessarily mean that the gas cost increases significantly over the expression itself, it just means that the stack item is duplicated and persisted when it gets use within its scope.
> and the new one would be SUB PUSHN jmpdest JUMP DUP1 PUSH0 LT PUSHN jmpdest JUMPI (jumpback logic based on that if statement...), which is not a clear improvement and may even cost more gas (I have of course omitted stack management for the other variables).

I decided to go ahead and make a simple gas test to just compare the two approaches on the average non-zero transfer amount. The version I recommended is indeed a little more expensive. Thank you for the guidance on how to think about these - will make a gas test in the future prior to recommending something like that!

> ceteris paribus I think the current version is a little more explicit about the behavior on different surplus values

Its now a moot point regardless but i do hear you here